lib/state-store.js

```diff
@@ -855,30 +855,54 @@ class StateStore {
   /**
    * Store a shard checkpoint.
    *
+   * @param {Date} approximateArrivalTimestamp - The approximate arrival timestamp.
    * @param {string} shardId - The ID of the shard to store a checkpoint for.
    * @param {string} checkpoint - The sequence number to store as the recovery point.
    * @param {string} shardsPath - The path pointing to where the shards state is stored.
    * @param {Object} shardsPathNames - The values of the attribute names in the path.
    * @fulfil {undefined}
    * @returns {Promise}
    */
-  async storeShardCheckpoint(shardId, checkpoint, shardsPath, shardsPathNames) {
+  async storeShardCheckpoint(
+    approximateArrivalTimestamp,
```

Comment:
Move `approxArrival` to the end of the arguments in this function. Order is by priority. Shard ID is the highest.